homeassistant/helpers/service.py

```diff
@@ -2,6 +2,10 @@
 import functools
 import logging
 
+# pylint: disable=unused-import,ungrouped-imports
+from typing import Optional  # NOQA
+from homeassistant.helpers.typing import HomeAssistantType  # NOQA
```

Comment:
Ok, so checking how many HomeAssistantType are optional

```
bootstrap.py:13:from homeassistant.helpers.typing import HomeAssistantType
bootstrap.py:221:                     hass: Optional[HomeAssistantType]=None,
Binary file matches: __pycache__/bootstrap.cpython-35.pyc
helpers/typing.py:27:    HomeAssistantType = NewType('HomeAssistantType', 'HomeAssistant')
helpers/typing.py:37:    HomeAssistantType = 'HomeAssistant'  # type: ignore
helpers/event_decorators.py:6:from homeassistant.helpers.typing import HomeAssistantType  # NOQA
helpers/event_decorators.py:10:HASS = None  # type: Optional[HomeAssistantType]
helpers/entity.py:15:from homeassistant.helpers.typing import HomeAssistantType
helpers/entity.py:28:                       hass: Optional[HomeAssistantType]=None) -> str:
helpers/entity.py:154:    hass = None  # type: Optional[HomeAssistantType]
helpers/service.py:7:from homeassistant.helpers.typing import HomeAssistantType  # NOQA
helpers/service.py:17:HASS = None  # type: Optional[HomeAssistantType]
helpers/script.py:8:from homeassistant.helpers.typing import ConfigType, HomeAssistantType
helpers/script.py:29:def call_from_config(hass: HomeAssistantType, config: ConfigType,
helpers/script.py:39:    def __init__(self, hass: HomeAssistantType, sequence, name: str=None,
components/recorder/__init__.py:20:                                          HomeAssistantType)
components/recorder/__init__.py:98:def setup(hass: HomeAssistantType, config: ConfigType) -> bool:
components/recorder/__init__.py:158:    def __init__(self, hass: HomeAssistantType,
```

So, not all of the usages of HomeAssistantType are optiona; I think it is better to leave that type as is.